src/xenia/base/main_win.cc

```diff
@@ -58,6 +58,16 @@ void AttachConsole() {
   fp = _fdopen(con_handle, "w");
   *stderr = *fp;
   setvbuf(stderr, nullptr, _IONBF, 0);
+
+  char* buf = nullptr;
```

Comment:
Why these changes?

Reply:
This is to make it work with a bash environment or Windows terminal environment. I looked at issue #1516 like you said to. I tried implement the solution to redirect stdout and stderr to the parent process, but this broke the bash terminal environment. So, I check if the SHELL environment variable exists. If it doesn't exists, I assume a Windows terminal environment.

MSYS2
![image](https://user-images.githubusercontent.com/11417992/88849030-44b2a480-d1b7-11ea-9093-816e5667cdaa.png)
![image](https://user-images.githubusercontent.com/11417992/88849056-5136fd00-d1b7-11ea-8db6-89d70ec2153b.png)

Windows Terminal
![image](https://user-images.githubusercontent.com/11417992/88849124-6dd33500-d1b7-11ea-88b9-363d8c74286d.png)

Reply:
However, if there is no console attached then you will get the message box pops up. For example, if you give invalid arguments in Visual Studio, then you will see the Message Box.

Reply:
What do you mean to make it work? You check with xe::has_console_attached if there's a console attached, so what does this specific piece of code in main_win.cc do? What happens or doesn't work if you remove it?

Reply:
xe::has_console_attched proves true if you launch from a Windows Terminal or bash environment. However, if you don't redirect stdout and stderr you won't see the print message in the Windows terminal, and Xenia will just exit silently like before. If you redirect stdout and stderr, then you won't see the message if Xenia is launched from a bash environment.  This code checks to see which environment Xenia was launched from and either redirects stdout and stderr, or doesn't redirect it.

So, in short, if you remove it, Xenia will still exit silently if launched from a Windows Terminal. 

I thought just getting a print message in the console would be better than a Message Box if xe::has_console_attached is true.

Reply:
If I can see your screenshot correctly, you're writing to a terminal which is not expecting output, which messes up the console for no reason.

As I said in #1516 , the windows Cmd and Powershell do not wait for output when launching Xenia through the command line. Instead of finding a quirky workaround to still output to a console which isn't expecting output, isn't it easier to fix has_console_attached to return false in those cases? (See AttachConsole in main_win.cc and feel free to improve it)

Reply:
Okay. I'll fix has_console_attached so that a Message Box will pop up if you are using the a Windows terminal.